Remotion/Web/Core/UI/Controls/WebTreeView.cs

```diff
@@ -669,8 +679,12 @@ private void RenderNodes (HtmlTextWriter writer, WebTreeNodeCollection nodes, bo
         bool isSelected,
         string nodePath,
         string nodeID,
-        int nestingDepth)
+        int nestingDepth,
+        bool isTopLevel)
     {
+      if (!isTopLevel && !string.IsNullOrEmpty (node.Category))
+        throw new InvalidOperationException ("Only root nodes may have a category set.");
```

Comment:
didn't see this in the earlier diff. we should move this to the callsite, solving the extra paramter issue. also, logically this check belongs to the callsite since it deals with grouping, not rendering of the node's head information.